src/components/Feed/Feed.jsx

```diff
@@ -0,0 +1,204 @@
+        console.error('postId is required for submitting a comment');
+        return;
+      }
+
+      const response = await fetch(`https://boolean-api-server.fly.dev/irlydo/post/${postId}/comment`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          content: newCommentContent,
+          contactId: 1, 
+          postId: postId
+        }),
+      });
+
+      const newComment = await response.json();
+
+      setCommentData((prevData) => [...prevData, newComment]);
+
+      setNewCommentContent('');
+    } catch (error) {
+      console.error('Error submitting comment:', error);
+    }
+  };
+
+  return (
+    <ul className="feed-ul">
+      <input
+        type='text'
+        placeholder="What's on your mind?"
+        value={newPostContent}
+        onChange={(e) => setNewPostContent(e.target.value)}
+      ></input>
+      <button type="button" className='submit-button' onClick={handlePostSubmit}>
+      <i className="fa-solid fa-paper-plane"></i>
+      </button>
+      {postData.map((post) => {
+        const contact = contactData.find((contact) => contact.id === post.contactId);
+        if (!contact) {
+          console.error(`Contact not found for post with id ${post.id}`);
+          return null;
+        }
+
+        let initials = `${contact.firstName[0]}${contact.lastName[0]}`;
+
+        const filteredComments = commentData
+        .filter((comment) => comment.postId === post.id)
+        .filter((comment) => !commentIdSet.has(comment.id));
+      
+        return (
+          <li key={post.id} className="post">
+            <div className="post-header">
+              <p className='initials'>{initials}</p>
+              <div className="author-and-title">
+                <h3 className="full-name">{contact.firstName} {contact.lastName}</h3>
+                <Link to={`/post/${post.id}`}><h5>{post.title}</h5></Link>
+              </div>
+            </div>
+            <p className='post-content'>{post.content}</p>
+            <hr></hr>
+            <ul>
+              {filteredComments.map((comment, index) => {
+                  const commenterContact = contactData.find((contact) => contact.id === comment.contactId);
+
+                  let commenterInitials = `${commenterContact.firstName[0]}${commenterContact.lastName[0]}`;
+
+                  const uniqueKey = `${post.id}-${index}`
+                  return (
+                    <li className="comment-section" key={uniqueKey}>
+                      <p className='commenter-initials'>{commenterInitials}</p>
+                      <div className="comment-name-and-content">
+                        <p className='commenter-name'>{commenterContact.firstName} {commenterContact.lastName}</p>
+                        <p>{comment.content}</p>
+                      </div>
+                    </li>
+                  );
+                })}
+            </ul>
```

Comment:
You'll probably need to break this down into smaller components for the individual parts and extract them out